### ISSUE TITLE
Argument precedence logic seems broken

### ISSUE THREAD

**Issue by thnee** [physera/onelogin-aws-cli#101]
## Expected Behavior

Setting `username` field in config file should result in it being used.

## Current Behavior

Setting `username` in config file appears to have no effect.

## Possible Solution

This is what I found:

- `cli._load_config` sets `username` to `None` in `cli_args`, which is correct because I did not specify any cli arg or environment variable.
- `cli.login` calls `config_section.set_overrides(vars(args))` which propagates `username=None` into the overrides.
- When `credentials.UserCredentials.load_username` calls `self.configuration['username']` it gets `None` because `configuration.Section.__get_item__` finds `username` in `self._overrides` which is `None`.

Not sure why go through the trouble of having this concept of "overrides". Seems like it would be simpler to just load all relevant cli args, environment variables, and config file fields at the start of the program, and then when it is time to read a value, you just get it from the first one available in that order.

## Steps to Reproduce

1. Specify username in config file.
2. Don't specify username in cli args or environment variable.
3. Observe in `__init__.get_saml_assertion` that `self.user_credentials.username` is `None` and onelogin returns http 400.

Tested with 0.1.9 and 0.1.10.

**Comment by slycoder** [comment]
Thanks for the report!  Do you have a (redacted) config file you can post here and I'll try to reproduce locally?

You might also want to try pulling master ... some config fixes have gone in recently which might've improved things.

**Comment by thnee** [comment]
In this shell session you can see that when setting username
via environment variable, it works as expected.

```
$ export ONELOGIN_AWS_CLI_PROFILE="redacted"
$ export ONELOGIN_AWS_CLI_USERNAME="mattias.lindvall@redacted.com"
$ export ONELOGIN_AWS_CLI_DURATION_SECONDS=3600

$ cat ~/.onelogin-aws.config
[defaults]
base_uri = https://api.eu.onelogin.com/
client_id = redacted
client_secret = redacted
subdomain = redacted
profile = redacted

[redacted-a]
aws_app_id = redacted

[redacted-b]
aws_app_id = redacted

$ onelogin-aws-login -C redacted-a
Onelogin Password:
Google Authenticator Token:
```

In this shell session you can see that when setting username
via config file, it breaks.

```
$ export ONELOGIN_AWS_CLI_PROFILE="redacted"
$ export ONELOGIN_AWS_CLI_DURATION_SECONDS=3600

$ cat ~/.onelogin-aws.config
[defaults]
base_uri = https://api.eu.onelogin.com/
client_id = redacted
client_secret = redacted
subdomain = redacted
profile = redacted
username = mattias.lindvall@redacted.com

[redacted-a]
aws_app_id = redacted

[redacted-b]
aws_app_id = redacted

$ onelogin-aws-login -C redacted-a
Onelogin Password:
Traceback (most recent call last):
  File "/home/ma0702180/.pyenv/versions/awscli/bin/onelogin-aws-login", line 11, in <module>
    load_entry_point('onelogin-aws-cli==0.1.10', 'console_scripts', 'onelogin-aws-login')()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/cli.py", line 55, in login
    api.save_credentials()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 131, in save_credentials
    self.assume_role()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 117, in assume_role
    self.get_role()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 103, in get_role
    self.get_arns()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 77, in get_arns
    self.get_saml_assertion()
  File "/home/ma0702180/.pyenv/versions/3.6.5/envs/awscli/lib/python3.6/site-packages/onelogin_aws_cli/__init__.py", line 58, in get_saml_assertion
    if saml_resp.mfa:
AttributeError: 'NoneType' object has no attribute 'mfa'
```

I debugged the code and found out that saml_resp is None
because the http response was 400 and username was None.

**Comment by slycoder** [comment]
Thanks.  I think both of the issues are resolved on master.  Can you verify and if so I can push out a new release to pip.

**Comment by slycoder** [comment]
Sorry for the delay but this fix is now in 1.11 on pypi.
